lib/internal/internal.go

```diff
@@ -346,7 +347,7 @@ func GenerateEmptyManifest(name string) (*schema.ImageManifest, error) {
 	}, nil
 }
 
-func GenerateManifestV22(name string, dockerURL *common.ParsedDockerURL, config *typesV2.ImageConfig, imageDigest string, lowerLayers []*schema.ImageManifest, debug log.Logger) (*schema.ImageManifest, error) {
+func GenerateManifestV22(name, manhash string, dockerURL *common.ParsedDockerURL, config *typesV2.ImageConfig, imageDigest string, lowerLayers []*schema.ImageManifest, debug log.Logger) (*schema.ImageManifest, error) {
```

Comment:
woah, this is getting immense ;-) at least we could make this a newline separated arglist and squash `imageDigest` with `name, manhash`. Also some documentation would welp. I am becoming concerned about the size of this arglist.